### ISSUE TITLE
in memory acked queue should scope the lock to the pipeline_id

### ISSUE THREAD

**Issue by ph** [elastic/logstash#6732]
Currently, when we run the spec we used an in memory acked queue, this testing queue will use the lock for the PQ. The problem is the created lock is a global lock and is not scoped to the `pipeline_id`, so creating more than one active pipelines will fail.

Also the created file for the lock is created in $LOGSTASH_HOME, if you start logstash and a test fails a **.lock** file could still be present in a future run since the **clear_data_dir** doesn't clear that file.

```ruby
def clear_data_dir
    data_path = agent_settings.get("path.data")
    Dir.foreach(data_path) do |f|
    next if f == "." || f == ".." || f == ".gitkeep"
    FileUtils.rm_rf(File.join(data_path, f))
  end
end
```

**Comment by colinsurprenant** [comment]
This is an awkward situation: memory acked queue shouldn't need external exclusive file locking because it does not uses files. But  memory acked queue are **now** used only for tests and it is important to mimic the persisted acked queue behaviour for the tests. 

So I suggest we still use file based exclusive file locking but instead of discarding the dir path of the persisted acked queue, we will use this same dir path to namespace the `.lock` file.

Note that an unused `.lock` file is not a problem and will not prevent a LS instance from starting if this `.lock` file has not been locked. In other words, just having a `.lock` file present does not make it into a locked state; it is not the file *presence* which does the locking.

**Comment by ph** [comment]
@colinsurprenant yup, +1 to make the differents queue behave the same, are we running the same tests against the two different queues type?

**Comment by colinsurprenant** [comment]
@ph the specs are run against the memory acked queue, minus a few specific test cases.

**Comment by colinsurprenant** [comment]
For reference, this is where the queue options are set for specs:  https://github.com/elastic/logstash/blob/9cc165a428a356c974d7ae56f4789cd77a6b3f74/spec/spec_helper.rb#L44-L46

**Comment by colinsurprenant** [comment]
fixed by #6739
